File: mobile/ios/jet-meal/jet-meal/JetMealWebViewStore.swift
//
//  JetMealWebViewStore.swift
//  jet-meal
//

import Combine
import WebKit

/// Позволяет из SwiftUI вызывать `reload` / `goBack` на созданном `WKWebView`.
final class JetMealWebViewStore: ObservableObject {
    weak var webView: WKWebView?

    var canGoBack: Bool {
        webView?.canGoBack ?? false
    }

    func reload() {
        webView?.reload()
    }

    func goBack() {
        webView?.goBack()
    }

    /// Вызывать после навигации, чтобы SwiftUI обновила доступность «Назад».
    func notifyNavigationChanged() {
        objectWillChange.send()
    }
}

File: mobile/android/app/src/main/java/ru/jet_meal/JetMealWebViewStore.kt
package ru.jet_meal

import android.webkit.WebView

/** Позволяет из Compose вызывать `reload` / `goBack` на созданном `WebView`. */
class JetMealWebViewStore {
    var webView: WebView? = null
        internal set

    val canGoBack: Boolean
        get() = webView?.canGoBack() == true

    fun reload() {
        webView?.reload()
    }

    fun goBack() {
        webView?.goBack()
    }

    var onNavigationChanged: (() -> Unit)? = null

    /** Вызывать после навигации, чтобы обновилась доступность «Назад». */
    fun notifyNavigationChanged() {
        onNavigationChanged?.invoke()
    }
}
